content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/troubleshooting.md

```diff
@@ -10,19 +10,157 @@ further_reading:
+
+### Manually verify injection in the application container
+
+If the Datadog UI does not show any instrumentation issues, or if you're troubleshooting a single service or container, you can manually verify whether injection occurred as expected. This method is helpful when debugging in environments where centralized visibility is limited or when a specific service isn't reporting traces.
+
+To confirm injection at the container level, check that:
+
+1. `/etc/ld.so.preload` includes the following entry: 
+   ```
+   /opt/datadog-packages/datadog-apm-inject/stable/inject/launcher.preload.so
+   ``` 
+2. The `LD_PRELOAD` environment variable is set to the same value.
+3. The directory `/opt/datadog-packages/datadog-apm-inject` exists, with `stable` and `$version` subdirectories.
+4. Language-specific directories exist (for example, `/opt/datadog/apm/library/java/` for Java).
+
+To enable debug logs:
+1. Set `DD_APM_INSTRUMENTATION_DEBUG: true` in the pod spec.
+2. Delete the pod to enable debug logs during injection.
+
+## Configuration options that affect injection
+
+There are several configuration mechanisms that can block or alter injection behavior.
+
+### Injector version
+
+To set the injector version:
+- At the cluster level:
+  
+  Set in `values.yaml` under `datadog.apm.instrumentation.injector.imageTag`.
+
+- At the pod level:
+ 
+  Set with the annotation `admission.datadoghq.com/apm-inject.version`.
+
+
+For host or Docker injection, modifying the `auto_inject` version is not recommended. 
+
+To manually update the injector version, run:
+```
+sudo datadog-installer install "oci://install.datadoghq.com/apm-inject-package: <VERSION>-1"
+```
+
+Alternatively, set the following in the installation script:
+```
+DD_INSTALLER_DEFAULT_PKG_VERSION_DATADOG_APM_INJECT=<VERSION>-1
+```
+
+### Allow and deny lists
+
+#### Default deny list
+
+Datadog maintains an internal deny list to prevent injection into certain processes (for example, IDEs or databases). If a process command or entrypoint is on this list, the injector exits the injection process.  
+
+#### Custom deny list entries (Linux only)  
+
+{{< callout url="https://docs.google.com/forms/d/e/1FAIpQLSdMu6WAsUCD3djkl_oN0Qh7fQmBCiKYyUvuqlYWRyObebAc6Q/viewform" header="Join the Preview!">}}
+Workload selection is available for Linux-based apps through a limited availability preview. To configure allow or deny rules for process injection, sign up for preview access.
+{{< /callout >}}
+
+#### Injection container flagged by security scanners
+
+Security tools may flag the `apm-inject` container because it runs an executable at startup, which can resemble malicious software. 
+
+The container's behavior is expected and safe; the executable configures the environment for auto-instrumentation.
+
+Datadog adheres to security best practices and is working with security vendors to allow this container.
```

Comment:
I would maybe push the "Injection container flagged by security scanners" section below "Kubernetes workload selection" since the latter is more closely related to "Custom deny list entries (Linux only)".

Reply:
I would also consider naming "Custom deny list entries (Linux only)" to "Linux workload selection" to make it obvious to users that these are the sections for the same goals but instructions for different deployments.